vembrane/modules/fhir.py

```diff
@@ -0,0 +1,198 @@
+
+# intervaltree is untyped, so we use type: ignore to suppress type checking errors
+from intervaltree import IntervalTree  # type: ignore
+
+from vembrane.common import add_common_arguments
+from vembrane.modules import structured
+
+PROFILE_DIR = (
+    importlib.resources.files("vembrane.modules") / "assets" / "fhir" / "profiles"
+)
+
+
+def add_subcommand(subparsers):
+    parser = subparsers.add_parser("fhir")
+    parser.add_argument(
+        "vcf",
+        help="The file containing the variants.",
+        nargs="?",
+        default="-",
+    )
+    parser.add_argument(
+        "sample",
+        help="The sample to use for generating FHIR output.",
+    )
+    parser.add_argument(
+        "assembly",
+        help="The reference assembly used for read mapping.",
+        choices=["GRCh38", "GRCh37"],
+    )
+    parser.add_argument(
+        "--url",
+        "-u",
+        help="Generic url used as identifier by FHIR e.g. http://<institute>/<department>/VCF",
+        default=None,
+    )
+    parser.add_argument(
+        "--status",
+        "-s",
+        help="Status of findings. E.g. final, preliminary, ...",
+        default=None,
+    )
+    parser.add_argument(
+        "--coordinates",
+        "-c",
+        help="Coordinate system of base positions.",
+        default=None,
+        choices=["0", "1"],
+    )
+    parser.add_argument(
+        "--profile",
+        required=True,
+        choices=[path.with_suffix("").name for path in PROFILE_DIR.iterdir()],
+        help="The FHIR profile to use for generating the output.",
+    )
+    parser.add_argument(
+        "--sample-allelic-frequency",
+        default=None,
+        help="Python expression calculating the the samples allelic frequency"
+        "as percentage. E.g. \"FORMAT['AF'][sample][0] * 100\" ",
+    )
+    parser.add_argument(
+        "--sample-allelic-read-depth",
+        default="FORMAT['AD'][sample][1]",
+        help="Python expression accessing the the samples allelic read depth."
+        "Default is: \"FORMAT['AD'][sample][1]\"",
+    )
+    parser.add_argument(
+        "--confidence-status",
+        default=None,
+        help="Python expression for calculating the variants confidence status being "
+        "High, Intermediate or Low. E.g. \"'High' if QUAL >= 20 else ('Intermediate' "
+        "if QUAL >= 10 else 'Low')\"",
+    )
+    parser.add_argument(
+        "--output-fmt",
+        choices=["json", "jsonl", "yaml"],
+        help="Output format. If not specified, can be automatically determined from "
+        "the --output file extension.",
+    )
+    parser.add_argument(
+        "--output",
+        "-o",
+        default="-",
+        help="Output file, if not specified, output is written to STDOUT.",
+    )
+    add_common_arguments(parser)
+
+
+def load_tsv(path, skip_comment=True):
+    with open(path, "r") as file:
+        reader = csv.reader(file, delimiter="\t")
+        if skip_comment:
+            next(reader)
```

Comment:
This only skips the first line, and since csv kind of allows comments (lines starting with "#") the parameter naming "skip_comment" is a bit misleading.